pkg/bgpv1/agent/controller.go

```diff
@@ -452,6 +459,13 @@ func (c *Controller) applyPolicyDefaults(policy *v2alpha1api.CiliumBGPPeeringPol
 func (c *Controller) validatePolicy(policy *v2alpha1api.CiliumBGPPeeringPolicy) error {
 	for _, r := range policy.Spec.VirtualRouters {
 		for _, n := range r.Neighbors {
+			switch {
+			case n.PeerPort == nil:
+				// This should not happen since the default value is set by k8s and applyPolicyDefaults().
+				return fmt.Errorf("undefined port for peer ASN %d IP %s", n.PeerASN, n.PeerAddress)
+			case *n.PeerPort < minPeerPort:
+				return fmt.Errorf("port must be greater than %d for peer ASN %d IP %s", minPeerPort-1, n.PeerASN, n.PeerAddress)
+			}
```

Comment:
Yes, the checks are not necessary since the API server validates, but the checks are here in case a test calls `Reconcile()`. I can remove if you have a strong opinion.